client/state/posts/test/utils.js

```diff
@@ -16,10 +16,80 @@ import {
 	getDeserializedPostsQueryDetails,
 	getSerializedPostsQueryWithoutPage,
 	getTermIdsFromEdits,
+	hasTermEditDifferences,
 	mergeIgnoringArrays
 } from '../utils';
 
 describe( 'utils', () => {
+	describe( 'hasTermEditDifferences', () => {
+		it( 'should return false if no term edits passed', () => {
+			expect( hasTermEditDifferences() ).to.be.false;
+		} );
+
+		it( 'should return false if term edits are the same as saved terms', () => {
+			const hasDifferences = hasTermEditDifferences( {
+				post_tag: [ 'ribs', 'chicken' ],
```

Comment:
I think I'm blanking on some normalization logic we implemented that I'm unable to find, but at what point does a post's `terms` turn from a plain object to an array of either objects or strings?